### PR TITLE
chore(ci): adds support for windows.

### PR DESCRIPTION
**Pull request by jcchavezs**
Ping @JBAhire

### CODE REVIEW

**Comment by JBAhire** [comment]
As per the documentation of circleci windows machine doesn't come with docker-compose pre-installed (ref: https://circleci.com/docs/2.0/hello-world-windows/#software-pre-installed-in-the-windows-image).

I have added step to install docker-compose using choco and it's working now! 🎉

**Review by JBAhire — DISMISSED** [review]

**Review by JBAhire — DISMISSED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,14 +1,18 @@
-version: 2
+version: 2.1
+
+orbs:
+  win: circleci/windows@2.2.0
 
 jobs:
-  e2e-test:
+  e2e-test-linux:
```
Comment (kotharironak) [inline .circleci/config.yml:7]:
```suggestion
  e2e-test-ubuntu:
```

File: .circleci/config.yml [hunk 2]
```diff
@@ -39,7 +73,8 @@ workflows:
               only:
                 - main
     jobs:
-      - e2e-test
+      - e2e-test-linux
   commit:
     jobs:
-      - e2e-test
+      - e2e-test-linux
```
Comment (kotharironak) [inline .circleci/config.yml:79]:
```suggestion
      - e2e-test-ubuntu
```

**Review by kotharironak — DISMISSED** [review]
lgtm

File: .circleci/config.yml (hunk 1 above)
Comment (JBAhire) [reply .circleci/config.yml:7]: @kotharironak , circleci calls it `Linux Machine` so I tried to follow similar naming scheme here.

Comment (jcchavezs) [reply .circleci/config.yml:7]: I'd rather keep it as linux because of what @JBAhire mentions.

File: .circleci/config.yml [hunk 3]
```diff
@@ -28,6 +32,36 @@ jobs:
...
             .circleci/tests.sh
 
+  e2e-test-windows:
+    executor: win/default
+
+    working_directory: ~/hypertrace
+    
+    steps:
+      - run:
+          name: "install docker-compose"
```
Comment (jcchavezs) [inline .circleci/config.yml:42]:
```suggestion
          name: "Install docker-compose"
```

**Review by JBAhire — APPROVED** [review]

**Pull request merged**